Test automated changelog

rogalmic opened the pull request:
A way to create a changelog automatically:
https://github.com/conventional-changelog/standard-version#commit-message-convention-at-a-glance

Only commits that follow the guide get to changelog...

rogalmic commented:
To test checkout this branch, add commits with correct format, then run:
` npm run release -- --release-as patch`
or
` npm run release -- --release-as minor`
or 
` npm run release -- --release-as major`
or
`npm run release -- --release-as 1.1.0`

wataash commented on package.json:
```diff
@@ -1,197 +1,199 @@
+      }
+    ]
+  }
 }
```
Finding out changes (`"release": "standard-version"` was added) of `package.json` was a bit of a chore, please split commits into *addition* and *generate* (or *re-indent* and *addition*) from the next time, if possible.

rogalmic replied:
Whoops :). I have not noticed that the file was completey rewritten while installing dependnecy. 

I will be more careful in future.

wataash commented:
For future maintainers:
`w=1` make this changes readable:
https://github.com/rogalmic/vscode-bash-debug/pull/63/files?w=1

wataash commented:
I suggest we always commit `package-lock.json` when it's updated, because this file describes current state of `node_modules`. It makes it possible to build project reproducibly or time-travel to previous build.
If you think we don't need this feature and consider always-commits as tedious, you can simply delete this file.

rogalmic commented:
I think the package.json did not update, otherwise i would commit it. But since I added dependency, it should have updated. I did not update the projects version yet, maybe that is the moment that dependencies are added to lock file?

wataash commented:
Hmm, `package-lock.json` has many additions:

```diff
$ git checkout package-lock.json
$ git diff --stat package-lock.json
$ npm install

> bash-debug@0.1.1 postinstall /home/wsh/src/vscode-bash-debug
> node ./node_modules/vscode/bin/install

Detected VS Code engine version: ^1.18.0
Found minimal version that qualifies engine range: 1.18.0
Fetching vscode.d.ts from: https://raw.githubusercontent.com/Microsoft/vscode/dcee2202709a4f223185514b9275aa4229841aa7/src/vs/vscode.d.ts
vscode.d.ts successfully installed!

npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.

> bash-debug@0.1.1 prepublish /home/wsh/src/vscode-bash-debug
> tsc -p ./src

up to date in 8.016s
$ git diff --stat package-lock.json
 package-lock.json | 1714 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 1702 insertions(+), 12 deletions(-)
$ git diff package-lock.json
diff --git a/package-lock.json b/package-lock.json
index f57ddee..0c9246c 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -16,6 +16,16 @@
                        "integrity": "sha512-diCxfWNT4g2UM9Y+BPgy4s3egcZ2qOXc0mXLauvbsBUq9SBKQfh0SmuEUEhJVFZt/p6UDsjg1s2EgfM6OSlp4g==",
                        "dev": true
                },
+               "JSONStream": {
+                       "version": "1.3.2",
+                       "resolved": "https://registry.npmjs.org/JSONStream/-/JSONStream-1.3.2.tgz",
+                       "integrity": "sha1-wQI3G27Dp887hHygDCC7D85Mbeo=",
+                       "dev": true,
+                       "requires": {
+                               "jsonparse": "1.3.1",
+                               "through": "2.3.8"
+                       }
+               },
                "abbrev": {
                        "version": "1.1.1",
                        "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
@@ -34,6 +44,34 @@
                                "json-schema-traverse": "0.3.1"
                        }
                },
+               "align-text": {
+                       "version": "0.1.4",
+                       "resolved": "https://registry.npmjs.org/align-text/-/align-text-0.1.4.tgz",
+                       "integrity": "sha1-DNkKVhCT810KmSVsIrcGlDP60Rc=",
...(snip)...
```

Can you try again the following commands to check `package-lock.json` has no diff?

```bash
touch package.json
npm install
git status --ignored
```

rogalmic commented:
In windows there is no touch, so i just added one space before first comma:
```powershell
PS C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug> git diff .\package.json
diff --git a/package.json b/package.json
index 8756f0e..7b09032 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "bash-debug",
+  "name": "bash-debug" ,
   "displayName": "Bash Debug",
   "version": "0.1.1",
   "publisher": "rogalmic",
(END)
PS C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug> npm install

> bash-debug@0.1.1 postinstall C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug
> node ./node_modules/vscode/bin/install

Detected VS Code engine version: ^1.18.0
Found minimal version that qualifies engine range: 1.18.0
Fetching vscode.d.ts from: https://raw.githubusercontent.com/Microsoft/vscode/dcee2202709a4f223185514b9275aa4229841aa7/src/vs/vscode.d.ts
vscode.d.ts successfully installed!


> bash-debug@0.1.1 prepublish C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug
> tsc -p ./src

PS C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug> git status --ignored
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   package.json

Ignored files:
  (use "git add -f <file>..." to include in what will be committed)

        node_modules/
        out/

no changes added to commit (use "git add" and/or "git commit -a")
PS C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug> npm --version
3.10.8
PS C:\Users\Michał\Documents\src\bash-debugger\vscode-bash-debug>
```

No idea why, but this package lock file does not get updated in this case. Maybe because this is a dev dependency?

rogalmic commented:
OK, so now I have newest npm installed and looks better, `npm install` actually updates the lock file. I guess i need to read more about this anyway :).

wataash commented:
I see. I'm using npm 5.7.1 (latest).
Maybe [npm release note](https://github.com/npm/npm/releases) makes it clear.
